app/models/medium.rb

```diff
@@ -295,11 +304,12 @@ def self.search_by(search_params, page)
     if search_params[:all_tags] == '1' && search_params[:tag_operator] == 'and'
       search_params[:tag_ids] = Tag.pluck(:id)
     end
-    admin = User.find_by(id: search_params[:user_id])&.admin?
+    user = User.find_by(id: search_params[:user_id])
     search = Sunspot.new_search(Medium)
     search.build do
       with(:sort, search_params[:types])
-      without(:sort, Medium.advanced_sorts) unless admin
+      without(:sort, 'RandomQuiz')
```

Comment:
So nobody can sort for "random quizzes" anymore?

Reply:
Nobody should be able to search Random Quizzes, not even admins. This restores previous behaviour. A 'RandomQuiz' is just a temporary guy: It is a quiz created by a random generator when a user starts a self test with random questions selected by choosing some tags. It is allowed to spend one day in the database until (see `medium.rb`)

```
  scope :expired, -> {
                    where(sort: 'RandomQuiz').where('created_at < ?', 1.day.ago)
                  }
```

and `lib/tasks/cleanup.rake` should make it go away... and hm, I thought there would be a cron job that executes this rake task (at least in production) periodically, but I can't seem to find it. That is one thing we should investigate (separate from this PR). When that rake tast was written we did not yet have `sidekiq-cron` running; with the current codebase a better solution than a rake task in a usual cron job would be a worker that is periodically scheduled via sidekiq. We have such things already happening, see `config/schedule.yml` and `app/workers`.